lib/assets/javascripts/turbolinks.js.coffee

```diff
@@ -136,8 +137,8 @@ changePage = (doc, options) ->
   triggerEvent EVENTS.BEFORE_UNLOAD
   document.title = title
 
-  swapNodes(targetBody, findNodes(document.body, '[data-turbolinks-temporary]'), keep: false)
```

Comment:
This is doing unnecessary work (firing unneeded `after-remove` events) when doing a full replacement, since the `body`, on which the swaps happen, is replaced below (L150).